src/helpers/assertStringsWithinText.ts

```diff
@@ -0,0 +1,34 @@
+import { ReactTestRendererNode } from 'react-test-renderer';
+
+export const assertStringsWithinText = (
+  rendererJSON: ReactTestRendererNode | Array<ReactTestRendererNode> | null
+) => {
+  if (!rendererJSON) return;
+
+  if (Array.isArray(rendererJSON)) {
+    rendererJSON.forEach(assertStringsWithinTextForNode);
+    return;
+  }
+
+  return assertStringsWithinTextForNode(rendererJSON);
+};
+
+const assertStringsWithinTextForNode = (node: ReactTestRendererNode) => {
+  if (typeof node === 'string') {
+    return;
+  }
+
+  if (node.type !== 'Text') {
+    node.children?.forEach((child) => {
+      if (typeof child === 'string') {
+        throw new Error(
+          `Text strings must be rendered within a host Text component. Text ${child} was rendered in a ${node.type}`
```

Comment:
```suggestion
          `Text strings must be rendered within a host Text component. Text "${child}" was rendered in a ${node.type}`
```

Reply:
Adding quotes should improve readability of the message.

Reply:
Maybe we should use the original wording of RN error: `Invariant Violation: Text strings must be rendered within a <Text> component`? and append a bit more details so that the final error message could be like:

```
Invariant Violation: Text strings must be rendered within a <Text> component.

Detected attempt to render "${child}" string within a <${node.type}> component.
```
Wdyt?

Reply:
I totally agree, I changed the error message and it looks very clear now, thanks ! 

<img width="1310" alt="Capture d’écran 2022-09-18 à 13 10 21" src="https://user-images.githubusercontent.com/64224599/190899381-dd38289f-19ce-46d1-94ed-067fb661cd5f.png">

However the multiline inline snapshots don't fit well in vs code so I replaced them with regular snapshots